pkg/segment/segexecution.go

```diff
@@ -209,6 +185,17 @@ func HelperQueryArithmeticAndLogical(queryOps []structs.QueryArithmetic, resMap
 	return &mresults.MetricsResult{Results: finalResult, State: mresults.AGGREGATED}
 }
 
+func setFinalRes(finalResult map[string]map[uint32]float64, groupID string, timestamp uint32, returnBool bool, comparisonBool bool, val float64) {
+	// If a comparison operator, return 0/1 rather than filtering.
+	if returnBool {
+		finalResult[groupID][timestamp] = 0
+		val = 1
+	}
+	if comparisonBool {
+		finalResult[groupID][timestamp] = val
+	}
+}
```

Comment:
This looks correct to me, but I think you can simplify the logic:
```
if returnBool {
    finalResult[groupID][timestamp] = comparisonBool
    return
}

// Filter based on the comparison.
if !comparisonBool {
    finalResult[groupID][timestamp] = 0
}
```

But if you think your version is easier to understand, you can keep your version